### ISSUE TITLE
[bug] timestamp on some extracted files is shifted +1 hour

### ISSUE THREAD

**Issue by Box333** [joz-k/ios_backup_extractor#9]
I just noticed, that the timestamp on some extracted files is shifted +1hour.. Trying to figure out why only some files are affected and not all.

**Comment by joz-k** [comment]
Is this about the difference from the Creation/Modification time stored in the Exif data? And is the difference exactly one hour? I can only speculate about the reason. The timestamp is based on the "Birth" date and "Last Modified" date stored in the binary plist metadata file for a given media file. If other iOS backup extractor solutions have the same problem, then this is an iOS bug (or feature). Maybe the daylight saving time difference or some other reason?

**Comment by Box333** [comment]
i extracted 1000 files and 560 had +1h shifted (i also comapared some of them to the original photo dates on  my iphone). it also depends of your local time. If you are in croatia and i am in france you have +1h time diffrence. maybe your code runs on your local time ?

**Comment by joz-k** [comment]
Yes, I think this issue may have something to do with a photo being taken in a different time zone. The binary plist I use to get the "Birth" and "LastStatusChange" dates for each photo stores these timestamps as a UTC epoch integer. There is no timezone encoded in this timestamp. It's just a number, like `1718373760`. Of course, I convert this timestamp from UTC to the user's local timezone. 
Can you please confirm that any photos with a shifted creation/modification time attribute were taken in a different time zone than you are in now?

**Comment by Box333** [comment]
no. they were all taken in  the same zone. maybe you didnt use 24h format ?

**Comment by joz-k** [comment]
There is no 12H/24H time handling involved when dealing with these timestamps. Also, a large number of photos would be affected in case of such a trivial error.
I'll keep this bug open. In case something new is discovered. It may also be that this is really an iOS bug.
